### PR TITLE
allow config file sopers to be specified by ?UID

### PR DESCRIPTION
**Pull request by jesopo**
and also allow config uid sopers to change accountname

### CODE REVIEW

File: libathemecore/privs.c [hunk 1]
```diff
@@ -265,6 +265,25 @@ soper_find_named(const char *name)
 
 	return NULL;
 }
+struct soper *
+soper_find_uid(const char *uid)
+{
+	struct soper *soper;
+	mowgli_node_t *n;
+	char lookup_uid[UIDLEN+1];
```
Comment (edk0) [inline libathemecore/privs.c:273]: I think this should be IDLEN+2

Comment (jesopo) [reply libathemecore/privs.c:273]: how come

Comment (jesopo) [reply libathemecore/privs.c:273]: oh null termination i forgot about that

Comment (edk0) [reply libathemecore/privs.c:273]: 2 for ? and null terminator, IDLEN because UIDLEN is for s2s UIDs

**Comment by jesopo** [comment]
I think this will currently not let you configure DB sopers by `?ID`. do we want that?

**Pull request merged**